src/NHibernate.Test/NHSpecificTest/GH1439/Entity.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace NHibernate.Test.NHSpecificTest.GH1439
+{
+	public class CompositeEntity
+	{
+		public virtual int Id { get; set; }
+		public virtual string Name { get; set; }
+		public virtual string LazyProperty { get; set; }
+
+		public override int GetHashCode()
+		{
+			return (Id + "|" + Name).GetHashCode();
+		}
+
+		public override bool Equals(object obj)
+		{
+			if (obj == null)
+				return false;
+
+			return obj.GetHashCode() == GetHashCode();
```

Comment:
Even if it is only a test, please use a correct implementation. Hashcode are not guaranteed unique per entity. Hashcode collisions occur and create bugs. Testing hashcode equality can only serves as a shortcut for early exit if they are unequal. If they are equal, the entities may still be different. So for such a simple case, better just test directly that both `Id` and `Name` are equals.